testplan/common/utils/helper.py

```diff
@@ -35,6 +38,58 @@
 from testplan.testing.multitest.result import Result
 
 
+class DriverLogCollector:
+
+    """
+    Customizable file collector class used for collecting driver logs.
+
+    :param name: Name of the object shown in the report.
+    :param description: Text description for the assertion.
+    :param ignore: List of patterns of file name to ignore when
+        attaching a directory.
+    :param file_pattern: List of patterns of file name to include when
+        attaching a directory. (Defaults: "stdout*", "stderr*")
+    :param recursive: Recursively traverse sub-directories and attach
+        all files, default is to only attach files in top directory.
+    :param failure_only: Only collect files on failure.
+    """
+
+    def __init__(
+        self,
+        name: str = "DriverLogCollector",
+        description: str = "logs",
+        ignore: list = None,
```

Comment:
Can we specify this more, like `List[str]` using `List` from `typing` module. Same goes for `file_pattern` parameter.